src/xpk/commands/job.py

```diff
@@ -0,0 +1,97 @@
+"""
+Copyright 2024 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+from ..core.commands import run_command_for_value
+from ..utils.console import xpk_exit, xpk_print
+from ruamel.yaml import YAML
+import re
+import sys
+
+
+def job_info(args):
+  """Run commands obtaining information about a job given by name.
+
+  Args:
+    args: user provided arguments for running the command.
+
+  Returns:
+    0 if successful and 1 otherwise.
+  """
+  job_name = args.name
+
+  desc_command = f'kubectl-kjob describe slurm {job_name}'
+  desc_code, desc_text = run_command_for_value(
+      desc_command, 'Getting job data', args
+  )
+  if desc_code != 0:
+    xpk_print(f'Data info request returned ERROR {desc_code}')
+    xpk_exit(desc_code)
+
+  job_command = (
+      'kubectl-kjob list slurm -o yaml --field-selector'
+      f' metadata.name=={job_name}'
+  )
+  job_code, job_text = run_command_for_value(
+      job_command, 'Getting job info', args
+  )
+  if job_code != 0:
+    xpk_print(f'Job info request returned ERROR {job_code}')
+    xpk_exit(job_code)
+
+  pods_command = f'kubectl get pods -l=job-name={job_name} --no-headers'
+  pods_code, pods_text = run_command_for_value(
+      pods_command, 'Getting pods list', args
+  )
+  if pods_code != 0:
+    xpk_print(f'Pods list request returned ERROR {pods_code}')
+    xpk_exit(pods_code)
+
+  yaml = YAML(typ='safe')
+  job_yaml = yaml.load(job_text)['items'][0]
+
+  output = {
+      'Job name': job_name,
+      'Profile': get_profile(job_yaml),
+      'Labels': job_yaml['metadata']['labels'],
+      'Mounts': job_yaml['spec']['template']['spec']['containers'][0][
+          'volumeMounts'
+      ],
+      'Environment variables': get_ev_vars(desc_text),
+      'Pods': get_pods(pods_text),
+  }
+
+  yaml.default_flow_style = False
+  yaml.sort_keys = False
+  yaml.dump(output, sys.stdout)
+
+  xpk_exit(0)
```

Comment:
I am not use if xpk_exit(0) is neccessary

Reply:
Right, I don't think it is. The code exits properly without it. Although some old commands (eg. cluster) add this at the end. Possible cleanup target.

Reply:
The only difference is the text prompt we get at the end ("Exiting XPK cleanly", "XPK Done.").